lib/plugins/package/lib/zipDirectory.js

```diff
@@ -1,16 +1,55 @@
 'use strict';
 
+const childProcess = require('child_process');
 const archiver = require('archiver');
 const BbPromise = require('bluebird');
 const path = require('path');
 const fs = require('fs');
 const globby = require('globby');
+const _ = require('lodash');
+const fileExistsSync = require('../../../utils/fs/fileExistsSync');
 
 module.exports = {
   zipDirectory(exclude, include, zipFileName) {
+    this.exclude = exclude;
+    this.include = include;
+    this.zipFileName = zipFileName;
+
+    return BbPromise.bind(this)
+      .then(this.excludeDevDependencies)
+      .then(this.zip);
+  },
+
+  excludeDevDependencies() {
+    // take care of node runtime-services
+    try {
+      const packageJsonFilePath = path.join(this.serverless.config.servicePath, 'package.json');
```

Comment:
I almost missed that one. If a serverless project has multiple functions packaged individually, it can well be that each function is developed in a nodejs sub-project. In such a project, this will refer to the `package.json` file of the root/parent project, which does not have the right dependencies.

We actually exactly do that: the serverless.yml is in a parent nodejs project that refers to nodejs "sub-projects", each having a separate `package.json` file. However there is no way to specify the root path of each subproject in the `serverless.yml` file, so maybe there is a need for a new property. Otherwise you will always filter the production dependencies of the parent project (usually just an empty reactor project used to build all the sub-projects). For example, we use https://github.com/lerna/lerna to do that.